command/agent/exec/server.go

```diff
@@ -153,8 +156,7 @@ func (s *Server) Run(ctx context.Context, envTmpls map[string]*config.EnvTemplat
 			s.runner.Stop()
 			s.childLock.Lock()
 			if s.child != nil {
-				// TODO: use kill to immediately kill or gracefully?
-				s.child.Kill()
+				s.child.Stop()
```

Comment:
Does this hard-kill the process after a timeout or do we need to do something special on top?

Reply:
https://pkg.go.dev/github.com/hashicorp/consul-template/child#Child.Stop
> Stop behaves almost identical to Kill except it suppresses future processes from being started by this child and it prevents the killing of the child process from sending its value back up the exit channel. This is useful when doing a graceful shutdown of an application.

_think_ we're good?